File: service/src/main/java/hu/bsstudio/safetycar/configuration/SafetyCarServiceConfiguration.java
package hu.bsstudio.safetycar.configuration;

import hu.bsstudio.repository.TeamRepository;
import hu.bsstudio.safetycar.DefaultSafetyCarService;
import hu.bsstudio.safetycar.SafetyCarService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class SafetyCarServiceConfiguration {

    @Autowired
    private TeamRepository repository;

    @Bean
    public SafetyCarService safetyCarService() {
        return new DefaultSafetyCarService(repository);
    }
}

File: server/service/src/main/kotlin/hu/bsstudio/robonaut/safetycar/configuration/SafetyCarServiceConfiguration.kt
package hu.bsstudio.robonaut.safetycar.configuration

import hu.bsstudio.robonaut.repository.TeamRepository
import hu.bsstudio.robonaut.safetycar.BroadcastingSafetyCarService
import hu.bsstudio.robonaut.safetycar.DefaultSafetyCarService
import hu.bsstudio.robonaut.safetycar.SafetyCarService
import org.springframework.amqp.rabbit.core.RabbitTemplate
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class SafetyCarServiceConfiguration(
    private val repository: TeamRepository,
    private val rabbitTemplate: RabbitTemplate,
) {
    @Bean
    fun safetyCarService(defaultSafetyCarService: SafetyCarService): SafetyCarService {
        return BroadcastingSafetyCarService(rabbitTemplate, defaultSafetyCarService)
    }

    @Bean
    fun defaultSafetyCarService(): SafetyCarService {
        return DefaultSafetyCarService(repository)
    }
}
